loopy/check.py

```diff
@@ -187,6 +187,27 @@ def check_for_integer_subscript_indices(kernel, callables_table):
                 type(insn).__name__))
 
 
+def check_sub_array_ref_inames_not_within_or_redn_inames(kernel):
+    all_within_inames = frozenset().union(*(insn.within_inames
+                                            for insn in kernel.instructions))
+    all_redn_inames = frozenset().union(*(insn.reduction_inames()
+                                          for insn in kernel.instructions))
+    all_sar_inames = frozenset().union(*(insn.sub_array_ref_inames()
+                                         for insn in kernel.instructions))
+
+    if all_sar_inames & all_within_inames:
+        sample = (all_sar_inames & all_within_inames).pop()
```

Comment:
```suggestion
        sample = next(iter(all_sar_inames & all_within_inames))
```
(Less stateful.)

Reply:
c9c46b9

Reply:
FWIW, it had correctness issues too as I was popping off of a frozenset. Thanks for the fix!